js/src/common/states/ModalManagerState.js

```diff
@@ -0,0 +1,88 @@
+import evented from '../utils/evented';
+
+import Modal from '../components/Modal';
+import ModalState from './ModalState';
+
+class ModalManagerState {
+  constructor() {
+    this.clear();
+  }
+
+  /**
+   * Show a modal dialog.
+   *
+   * @param {Modal} component
+   * @public
+   */
+  show(modalClass, attrs) {
+    const modal = new ModalState(modalClass, attrs);
+
+    if (!(modal.getClass().prototype instanceof Modal)) {
+      throw new Error('The ModalManager component can only show Modals');
+    }
```

Comment:
Is that supposed to intentionally break old code (expecting to call `show` on the `ModalManager` component), to make developers aware they need to upgrade?

Reply:
This was in Modal Manager's show before. That being said, we could either do this, put in a BC layer (if a modal component is sent in, package it's constructor and props in a modal POJO), or both?

Reply:
I think we should check there's a nice error message for the breaking change, like "must pass a class and not an instance". We mostly agreed not trying to do BC I believe.

Reply:
After doing a bit of digging it seems there is no "real" way of checking whether a variable contains an object or a class given how javascript works.

The `protoype instanceof` check above will work against both an object instance or the class itself.

I really think we need an error message for instances as this will be common with code updates.

It seems like we could check for the presence of one of the known methods.

So I'd suggest:

    if (!(type.prototype instanceof Modal)) {
      throw new Error('The ModalManager can only show Modals');
    }
    if (type.init) {
      throw new Error('The type parameter must be a modal class, not a modal instance');
    }